src/job.ts

```diff
@@ -193,14 +194,26 @@ export class CronJob<OC extends CronOnCompleteCommand | null = null, C = null> {
 		return this.cronTime.sendAt();
 	}
 
-	fireOnTick() {
-		for (const callback of this._callbacks) {
-			void callback.call(
-				this.context,
-				this.onComplete as WithOnComplete<OC> extends true
-					? CronOnCompleteCallback
-					: never
-			);
+	async fireOnTick() {
+		if (this.isRunning) {
```

Comment:
This breaks existing behavior and should be controlled by an option on the CronJob declaration (`waitForCompletion` or something), which would be set to `false` by default.

Reply:
Thank you for your review! Is my understanding correct if we add `waitForCompletion` as a parameter?

- waitForCompletion: `false` (default)
  - Maintains existing behavior
  - Starts new tasks regardless of previous task execution status

- waitForCompletion: `true`
  - Waits until previous task is completed
  - Does not start new tasks while `isRunning` is true